### DISCUSSION TITLE
Unable to bind CollectionView to Dictionary<string, object>

### DISCUSSION THREAD

**Discussion by daltzctr** [dotnet/maui#15249] [Q&A]
The following CollectionView binding where Items is of type `Dictionary<string, myobject>` fails with an exception.

```
<CollectionView ItemsSource="{Binding Items}">
   <datatemplatestuff/>
</CollectionView>
```

Exception:

```
Error: Cannot get 'Item' value (type 'Object') from type 'Microsoft.Maui.Controls.Platform.ItemTemplateContext'. BindingExpression: Path='Item' DataItem='Microsoft.Maui.Controls.Platform.ItemTemplateContext'; target element is 'Microsoft.Maui.Controls.Platform.ItemContentControl' (Name='ItemContentControl'); target property is 'FormsDataContext' (type 'Object').
```

I have no idea why this is the case (it previously worked in Xamarin) nor how to debug it.

**Comment by daltzctr** [comment]
For information, MAUI does not like binding to generics.

**Comment by janseris** [comment]
If this worked in Xamarin, report it as a bug in MAUI issues.

**Comment by WisJamison** [comment]
Hi @daltzctr Have you figured out a workaround for this issue? I ran into same issue and opened a ticket to Maui Team. #19208
